### PR TITLE
Tweaks cryostylane to use the new reaction mechanics, incorporating stasis iceblocks, freezing, temporary organ decay prevention and surgery speed modifiers.

### PR DESCRIPTION
**Pull request by Thalpy**
## About The Pull Request

Tweaks cryostylane to use the new reaction mechanics, incorporating stasis iceblocks, freezing, temporary organ decay prevention and surgery slows.

### Standard reagent mechanics

When applied to a dead (or living) mob it will slow down surgery steps by making them take +10% to +30% the time (depending on the creation purity, of course!)

Will tint them blue

Will pause organ decay

(To note, however, this reagent DOES process in the dead, and will be consumed over time, thus formaldehyde is better for longer term applications, or any patients that need a lot of surgery)

### Inverse reagent mechanics <50% purity

Freezes you up into a block of ice where you can't be touched until the chem leaves your system. Puts you in stasis while you're in there too.

### Reaction mechanics

Making impure reagent purely by pH is difficult with this reagent, you need to chill it below 300K. The colder it is, the larger the effect on impurity it has.

If it is dropped below 50K it will freeze up the area around it.

If your purity is below 20%, it will freeze up a bigger area and spawn a snowman.

### Tweaks the the base chem

Temperature changes with cryostylane are no longer instant, instead it adjusts temperature by -50K per second (5u = -50K). (If this breaks anything that anyone uses it for let me know and I'll figure out a solution).

## Why It's Good For The Game

Adds an optional, and worse, method to deal with decaying mobs. Since computers next to stasis beds are disappearing I figured this might help with the concerns over decaying bodies everywhere.
Adds more depth to organ decay prevention options.

## Changelog
:cl:
qol: Tweaks cryostylane to use the new reaction mechanics, incorporating stasis iceblocks, freezing, temporary organ decay prevention and surgery slows. Cryostylane when added to a corpse will tint them blue and pause organ decay temporarily, while it is in a mob it will slow down all surgery applied upon them. Inverse cryostylane will freeze you in a block of ice and put you in stasis (living and dead).
qol: Tweaks cyrostylane to have a more indepth reaction, where colder temperatures will cause it's purity to decrease.
code: cleans up some of holder.dm code and adds a REAGENT_IGNORE_STASIS bitflag.
/:cl:

### CODE REVIEW

**Comment by Iamgoofball** [comment]
Just add this behavior to the existing cold reagent from goonchem.

**Comment by Zytolg** [comment]
Freezes-The-Lizards
Can't wait for the inevitable Cryosenium snowball fights

**Comment by Thalpy** [comment]
> 
> 
> Just add this behavior to the existing cold reagent from goonchem.

What's the name of it?

File: code/datums/status_effects/neutral.dm [hunk 1]
```diff
@@ -342,3 +342,35 @@
...
+	duration = 600 //Will remove self when reagent is processed, though limited to 60s to be safe. I suppose if you're freezing the dying you want time but I'd rather have this incase oops 1000u
+	alert_type = /obj/screen/alert/status_effect/freon/stasis/cryosenium
+	var/obj/structure/ice_stasis/cube
+
+/datum/status_effect/cryosenium/on_apply()
+	cube = new /obj/structure/ice_stasis(get_turf(owner))
+	cube.color = "#03dbfc"
+	cube.anchored = 1
+	owner.forceMove(cube)
+	owner.status_flags |= GODMODE
```
Comment (Tlaltecuhtli) [inline code/datums/status_effects/neutral.dm:365]: hmmm

Comment (Fikou) [reply code/datums/status_effects/neutral.dm:365]: oh no no no

File: code/datums/status_effects/neutral.dm [hunk 2]
```diff
@@ -342,3 +342,35 @@
...
+* Freezes the player in a block of ice
+* Will be removed when the required reagent is removed too
+*/
+/obj/screen/alert/status_effect/freon/stasis/cryosenium
+	desc = "You're frozen inside of a protective ice cube! While inside, you can't do anything, but are immune to harm! You will be free when the chem runs out."
+
+/datum/status_effect/cryosenium
+	id = "cryosenium"
+	status_type = STATUS_EFFECT_UNIQUE
+	duration = 600 //Will remove self when reagent is processed, though limited to 60s to be safe. I suppose if you're freezing the dying you want time but I'd rather have this incase oops 1000u
```
Comment (Fikou) [inline code/datums/status_effects/neutral.dm:356]: use defines

**Comment by Maurukas** [comment]
>double to triple the time 

This is really quite significant in the context of surgery speed multipliers. This makes using this drug worse than doing surgery on the floor (50% speed). Surgery at 33% normal speed would make this the largest penalty possible in game by a large margin. Would you consider toning this down to something more along the lines of a ~15-25% speed penalty? That would still be very noticeable.

The main multipliers are in surgery_helpers.dm for reference.

**Comment by Thalpy** [comment]
> 
> 
> > double to triple the time
> 
> This is really quite significant in the context of surgery speed multipliers. This makes using this drug worse than doing surgery on the floor (50% speed). Surgery at 33% normal speed would make this the largest penalty possible in game by a large margin. Would you consider toning this down to something more along the lines of a ~15-25% speed penalty? That would still be very noticeable.
> 
> The main multipliers are in surgery_helpers.dm for reference.

Sure! I can make it less dramatic, give me a moment.

File: code/datums/status_effects/neutral.dm (hunk 1 above)
Comment (Thalpy) [reply code/datums/status_effects/neutral.dm:365]: It partially borrows from the other ice cube effect, is there an alternative that would be better?

File: code/datums/status_effects/neutral.dm (hunk 2 above)
Comment (Thalpy) [reply code/datums/status_effects/neutral.dm]: Done!

**Comment by TheBonded** [comment]
Will there be an opposite to this that decays organs faster but makes the body easier to do surgery on?

**Comment by Thalpy** [comment]
Alright I made it adjust speed by 10-30% (purity of 1 to 0.5)



> 
> 
> Will there be an opposite to this that decays organs faster but makes the body easier to do surgery on?

Hmm, I really like that idea. I'll think it over and see if I can find a good implementation - I might use that to create a competitive reaction since I haven't found a good use of those yet.

File: code/datums/status_effects/neutral.dm (hunk 1 above)
Comment (Tlaltecuhtli) [reply code/datums/status_effects/neutral.dm:365]: to do what

**Comment by Maurukas** [comment]
Thank you. The nature of surgery means that relatively tiny multipliers have a rather large effect.

There is a chem, sterilizine, which is _supposed_ to make surgery faster, though having used it quite a bit I am convinced that it is bugged and does nothing. Mothblocks may still disagree with me.

The ice cube behavior Goof is referring to is on the chem Cryostylane. In the original goonchem, this drug can freeze a mob into an ice cube, in addition to reacting with oxygen to lower temperatures in its container. When this chem was implemented in tg, only the reagent temperature behavior was implemented, and the ice cube behavior was not.

**Comment by Iamgoofball** [comment]
Ye, ice cubing got added with Freon

**Comment by TheBonded** [comment]
> > > double to triple the time
> > 
> > 
> > This is really quite significant in the context of surgery speed multipliers. This makes using this drug worse than doing surgery on the floor (50% speed). Surgery at 33% normal speed would make this the largest penalty possible in game by a large margin. Would you consider toning this down to something more along the lines of a ~15-25% speed penalty? That would still be very noticeable.
> > The main multipliers are in surgery_helpers.dm for reference.
> 
> Sure! I can make it less dramatic, give me a moment.

Don't forget to update the pr to say 10-30 instead of the current double to triple it still says

**Comment by Thalpy** [comment]
> 
> 
> Thank you. The nature of surgery means that relatively tiny multipliers have a rather large effect.
> 
> There is a chem, sterilizine, which is _supposed_ to make surgery faster, though having used it quite a bit I am convinced that it is bugged and does nothing. Mothblocks may still disagree with me.
> 
> The ice cube behavior Goof is referring to is on the chem Cryostylane. In the original goonchem, this drug can freeze a mob into an ice cube, in addition to reacting with oxygen to lower temperatures in its container. When this chem was implemented in tg, only the reagent temperature behavior was implemented, and the ice cube behavior was not.

Thanks for giving me feedback! Sometimes it's hard to get a good grasp of number ranges.

> 
> 
> Ye, ice cubing got added with Freon

I'll see if I can merge it with Cryostylane. Are there any cases where you would want to use the temperature mechanics inside of a mob? Though I suppose it doesn't really stop that since it'll just make any surgery done on you slower.

**Comment by TheBonded** [comment]
I don't quite understand how you get frozen in a block of ice with this, is it on injection? Would you mind explaining for me?

**Comment by Thalpy** [comment]
> 
> 
> I don't quite understand how you get frozen in a block of ice with this, is it on injection? Would you mind explaining for me?

If the purity of the reagent is below 50%, when it's added to a mob (either via injection, pill, or patch) it will turn into it's associated inverse chem, which is Cyrogelidia aka `/datum/reagent/inverse/cryosenium` This then calls the on_mob_add proc for that, which add a status effect (though, thinking about it now, it probably doesn't have to be in a status effect, I just borrowed stuff from freon). The cube puts the mob inside of it and prevents you from being harmed for the time being (though I might try to swap godmode with stasis, and add a IGNORE_STASIS bitflag for reagents.). The ice cube effect is removed when the reagent leaves the mob - or after 60s (just in case someone tries to put 1000u into someone _somehow_ and they get stuck for several minutes.)

**Comment by TheBonded** [comment]
> > I don't quite understand how you get frozen in a block of ice with this, is it on injection? Would you mind explaining for me?
> 
> If the purity of the reagent is below 50%, when it's added to a mob (either via injection, pill, or patch) it will turn into it's associated inverse chem, which is Cyrogelidia aka `/datum/reagent/inverse/cryosenium` This then calls the on_mob_add proc for that, which add a status effect (though, thinking about it now, it probably doesn't have to be in a status effect, I just borrowed stuff from freon). The cube puts the mob inside of it and prevents you from being harmed for the time being (though I might try to swap godmode with stasis, and add a IGNORE_STASIS bitflag for reagents.). The ice cube effect is removed when the reagent leaves the mob - or after 60s (just in case someone tries to put 1000u into someone _somehow_ and they get stuck for several minutes.)

Stasis would make this chemical used purposely with less purity to turn operating tables into stasis beds due to the recent operating computer changes to stasis beds. Might be good to add that to the main chem, but make it harder to acquire?

**Comment by D4C-420** [comment]
> If it is dropped below 50K it will explode, freezing up the area around it.

when you say explode do you mean like potassium+ water with everything freezing or do you mean just turning the floors to ice

**Comment by Mothblocks** [comment]
@Maurukas Sterilizine definitely *works*, it just sucks 😄

File: code/datums/status_effects/neutral.dm [hunk 3]
```diff
@@ -342,3 +342,35 @@
...
+	id = "cryosenium"
+	status_type = STATUS_EFFECT_UNIQUE
+	duration = 60 SECONDS //Will remove self when reagent is processed, though limited to 60s to be safe. I suppose if you're freezing the dying you want time but I'd rather have this incase oops 1000u
+	alert_type = /atom/movable/screen/alert/status_effect/freon/cryosenium
+	var/obj/structure/ice_stasis/cube
+
+/datum/status_effect/cryosenium/on_apply()
+	cube = new /obj/structure/ice_stasis(get_turf(owner))
+	cube.color = "#03dbfc"
+	cube.anchored = 1
```
Comment (Mothblocks) [inline code/datums/status_effects/neutral.dm:363]:
```suggestion
	cube.anchored = TRUE
```

File: code/datums/status_effects/neutral.dm [hunk 4]
```diff
@@ -342,3 +342,35 @@
...
+	owner.status_flags |= GODMODE
+	return ..()
+
+/datum/status_effect/cryosenium/tick()
+	if(!cube || owner.loc != cube || !owner.reagents.has_reagent(/datum/reagent/inverse/cryosenium)) //mostly a just in case kinda thing.
+		owner.remove_status_effect(src)
+
+/datum/status_effect/cryosenium/on_remove()
+	if(cube)
+		qdel(cube)
```
Comment (Mothblocks) [inline code/datums/status_effects/neutral.dm:374]:
```suggestion
	QDEL_NULL(cube)
```

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 5]
```diff
@@ -1436,3 +1436,38 @@
...
+
+/datum/reagent/medicine/cryosenium/on_mob_add(mob/living/consumer, amount)
+	. = ..()
+	consumer.mob_surgery_speed_mod = (0.4 * (1 - creation_purity))+0.1 //10% - 30% slower
+	consumer.color = "#03f4fc"
+
+/datum/reagent/medicine/cryosenium/on_mob_delete(mob/living/consumer)
+	. = ..()
+	consumer.mob_surgery_speed_mod = 1
+	consumer.color = "#ffffff"
```
Comment (Mothblocks) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm:1463]: Use the color defines.

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 6]
```diff
@@ -1436,3 +1436,38 @@
...
+	color = "#03f4fc"
+	impure_chem = null
+	inverse_chem_val = 0.5
+	inverse_chem = /datum/reagent/inverse/cryosenium
+	failed_chem = null
+	chemical_flags = REAGENT_DEAD_PROCESS
+
+/datum/reagent/medicine/cryosenium/on_mob_add(mob/living/consumer, amount)
+	. = ..()
+	consumer.mob_surgery_speed_mod = (0.4 * (1 - creation_purity))+0.1 //10% - 30% slower
```
Comment (Mothblocks) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm:1457]: Extract 0.4 and 0.1 to defines to make this equation easier to read.

File: code/modules/reagents/chemistry/recipes/medicine.dm [hunk 7]
```diff
@@ -241,3 +241,56 @@
...
+	purity_min = 0.15
+	reaction_flags = REACTION_HEAT_ARBITARY
+
+//purity != temp (above 50) - the colder you are the more impure it becomes
+/datum/chemical_reaction/cryosenium/reaction_step(datum/equilibrium/reaction, datum/reagents/holder, delta_t, delta_ph, step_reaction_vol)
+	. = ..()
+	if(holder.chem_temp < 50)
+		overheated(holder, reaction, step_reaction_vol)
+		return
+	var/datum/reagent/medicine/cryosenium/cryo = holder.has_reagent(/datum/reagent/medicine/cryosenium)
```
Comment (Mothblocks) [inline code/modules/reagents/chemistry/recipes/medicine.dm:268]: Name this cryosenium here and everywhere else, it'll make it more obvious and not conflict with anything else cryo.

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 8]
```diff
@@ -1436,3 +1436,38 @@
...
+	consumer.mob_surgery_speed_mod = 1
+	consumer.color = "#ffffff"
+
+//Pauses decay! Does do something, I promise.
+/datum/reagent/medicine/cryosenium/on_mob_dead(mob/living/carbon/consumer)
+	. = ..()
+	metabolization_rate = 0.05 * REM
+
+/datum/reagent/medicine/cryosenium/on_mob_life(mob/living/carbon/consumer)
+	if(consumer.stat > DEAD)
```
Comment (Mothblocks) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm:1471]: This isn't possible, did you mean `< DEAD`?

File: code/modules/reagents/chemistry/recipes/medicine.dm [hunk 9]
```diff
@@ -241,3 +241,56 @@
...
+	if(!cryo)
+		return
+	var/step_temp = ((holder.chem_temp-50)/250)
```
Comment (Mothblocks) [inline code/modules/reagents/chemistry/recipes/medicine.dm:271]: Move these out to a define as well (and likely change the `< 50` part to use it as well).

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Comment by Thalpy** [comment]
> 
> 
> > If it is dropped below 50K it will explode, freezing up the area around it.
> 
> when you say explode do you mean like potassium+ water with everything freezing or do you mean just turning the floors to ice

Oh, sorry, I don't mean a destructive explosion, it's just a habit to call something an explosion. When it's underheated (referred to in code as overheated) it will freeze the local area, no explosions. I'll see if I can clear that up in the PR text.

File: code/modules/reagents/chemistry/reagents/impure_reagents.dm [hunk 10]
```diff
@@ -51,3 +51,28 @@
 	ph = 1.5
 	taste_description = "an awful, strongly chemical taste"
 	color = "#270d03"
+
+//Freezes you in a block of ice for a small amount of time, 1s = 1u. Still does the
```
Comment (Mothblocks) [inline code/modules/reagents/chemistry/reagents/impure_reagents.dm:55]: Still does the...?

**Comment by D4C-420** [comment]
> Oh, sorry, I don't mean a destructive explosion, it's just a habit to call something an explosion. When it's underheated (referred to in code as overheated) it will freeze the local area, no explosions. I'll see if I can clear that up in the PR text.

thanks. was not really clear on that.

File: code/datums/status_effects/neutral.dm (hunk 1 above)
Comment (Thalpy) [reply code/datums/status_effects/neutral.dm:365]: It's alright, I'm converting it to stasis instead.

File: code/modules/reagents/chemistry/reagents/impure_reagents.dm (hunk 10 above)
Comment (Thalpy) [reply code/modules/reagents/chemistry/reagents/impure_reagents.dm]: Oops, got distracted

File: code/modules/reagents/chemistry/recipes/medicine.dm (hunk 9 above)
Comment (Thalpy) [reply code/modules/reagents/chemistry/recipes/medicine.dm]: Done

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm (hunk 8 above)
Comment (Thalpy) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: Yes, but on_mob_life can't be called while alive so it's a pointless check

File: code/modules/reagents/chemistry/recipes/medicine.dm (hunk 7 above)
Comment (Thalpy) [reply code/modules/reagents/chemistry/recipes/medicine.dm]: Done

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm (hunk 6 above)
Comment (Thalpy) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: Done

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm (hunk 5 above)
Comment (Thalpy) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: Done

File: code/datums/status_effects/neutral.dm (hunk 4 above)
Comment (Thalpy) [reply code/datums/status_effects/neutral.dm]: Done

File: code/datums/status_effects/neutral.dm (hunk 3 above)
Comment (Thalpy) [reply code/datums/status_effects/neutral.dm]: Oldcode strikes again!

File: code/datums/status_effects/neutral.dm (hunk 1 above)
Comment (Thalpy) [reply code/datums/status_effects/neutral.dm]: Okay it's stasis now

File: code/modules/reagents/chemistry/holder.dm [hunk 11]
```diff
@@ -700,6 +713,53 @@
...
 
+/*Processes the reagents in the holder and converts them, only called in a mob/living/carbon on addition
+*
+* Arguments:
+* * reagent - the added reagent datum/object
+* * added_volume - the volume of the reagent that was added (since it can already exist in a mob)
+* * added_purity - the purity of the added volume
+*/
+/datum/reagents/proc/process_mob_reagent_purity(_reagent, added_volume, added_purity)
+	var/datum/reagent/R = has_reagent(_reagent)
```
Comment (Mothblocks) [inline code/modules/reagents/chemistry/holder.dm:724]: better variable name

Comment (Thalpy) [reply code/modules/reagents/chemistry/holder.dm]: Pirates removed

**Comment by Thalpy** [comment]
> 
> 
> Thank you. The nature of surgery means that relatively tiny multipliers have a rather large effect.
> 
> There is a chem, sterilizine, which is _supposed_ to make surgery faster, though having used it quite a bit I am convinced that it is bugged and does nothing. Mothblocks may still disagree with me.
> 
> The ice cube behavior Goof is referring to is on the chem Cryostylane. In the original goonchem, this drug can freeze a mob into an ice cube, in addition to reacting with oxygen to lower temperatures in its container. When this chem was implemented in tg, only the reagent temperature behavior was implemented, and the ice cube behavior was not.

From code diving it increases the surgery speed by 20%, if and only if, you have a surgery prepared, and it will only work on that one surgery.

File: code/modules/reagents/chemistry/holder.dm [hunk 12]
```diff
@@ -700,6 +714,53 @@
...
+		if(!(reagent.chemical_flags & REAGENT_SPLITRETAINVOL))
+			remove_reagent(reagent.type, impureVol, FALSE)
+		add_reagent(reagent.impure_chem, impureVol, FALSE, added_purity = 1-reagent.creation_purity)
+	reagent.chemical_flags |= REAGENT_DONOTSPLIT
+
+///Processes any chems that have the REAGENT_IGNORE_STASIS bitflag ONLY
+/datum/reagents/proc/handle_stasis_chems(mob/living/carbon/owner, delta_time, times_fired)
+	var/need_mob_update = FALSE
+	for(var/datum/reagent/reagent as anything in reagent_list)
+		if(!reagent.chemical_flags & REAGENT_IGNORE_STASIS)
```
Comment (Mothblocks) [inline code/modules/reagents/chemistry/holder.dm:756]:
```suggestion
		if(!(reagent.chemical_flags & REAGENT_IGNORE_STASIS))
```

File: code/modules/reagents/chemistry/equilibrium.dm [hunk 13]
```diff
@@ -379,10 +379,10 @@
 
 	//Apply thermal output of reaction to beaker
 	if(reaction.reaction_flags & REACTION_HEAT_ARBITARY)
-		holder.chem_temp += (reaction.thermic_constant* total_step_added*thermic_mod) //old method - for every bit added, the whole temperature is adjusted
+		holder.chem_temp += clamp((reaction.thermic_constant* total_step_added*thermic_mod), 0, 99999) //old method - for every bit added, the whole temperature is adjusted
```
Comment (Mothblocks) [inline code/modules/reagents/chemistry/equilibrium.dm:382]: Where does 99999 come from? Can you make it a define, here and in 385?

File: code/modules/reagents/chemistry/holder.dm [hunk 14]
```diff
@@ -653,39 +654,52 @@
...
 	if(owner && need_mob_update) //some of the metabolized reagents had effects on the mob that requires some updates.
 		owner.updatehealth()
 		owner.update_stamina()
 	update_total()
 
+/*
+ * Metabolises a single reagent for a target owner carbon mob. See above.
+ *
+ * Arguments:
+ * * mob/living/carbon/C - The mob to metabolize in, if null it uses [/datum/reagents/var/my_atom]
```
Comment (Mothblocks) [inline code/modules/reagents/chemistry/holder.dm:669]:
```suggestion
 * * mob/living/carbon/owner - The mob to metabolize in, if null it uses [/datum/reagents/var/my_atom]
```

Comment (Mothblocks) [reply code/modules/reagents/chemistry/holder.dm:669]: Also, you have other arguments, you should document those too.

File: code/modules/reagents/chemistry/holder.dm [hunk 15]
```diff
@@ -653,39 +654,52 @@
...
+/datum/reagents/proc/metabolize_reagent(mob/living/carbon/owner, datum/reagent/reagent, delta_time, times_fired, can_overdose = FALSE, liverless = FALSE)
+	var/need_mob_update = FALSE
+	if(QDELETED(reagent.holder))
+		return FALSE
+
+	if(!owner)
+		owner = reagent.holder.my_atom
+
+	if(owner && reagent)
+		if(owner.reagent_check(reagent, delta_time, times_fired) != TRUE)
```
Comment (Mothblocks) [inline code/modules/reagents/chemistry/holder.dm:682]: Early returns.

File: code/modules/reagents/chemistry/holder.dm [hunk 16]
```diff
@@ -1335,7 +1360,7 @@
 		return
 
 	. = chem_temp
-	chem_temp = _temperature
+	chem_temp = clamp(_temperature, 0, 99999)
```
Comment (Mothblocks) [inline code/modules/reagents/chemistry/holder.dm:1363]: This should probably use the same define (99999).

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/modules/reagents/chemistry/holder.dm (hunk 15 above)
Comment (Thalpy) [reply code/modules/reagents/chemistry/holder.dm:682]: What do you mean?

File: code/modules/reagents/chemistry/equilibrium.dm (hunk 13 above)
Comment (Thalpy) [reply code/modules/reagents/chemistry/equilibrium.dm]: Arbitarily, but I made it a define

File: code/modules/reagents/chemistry/holder.dm (hunk 15 above)
Comment (Thalpy) [reply code/modules/reagents/chemistry/holder.dm]: Ah I see

File: code/modules/reagents/chemistry/holder.dm (hunk 14 above)
Comment (Thalpy) [reply code/modules/reagents/chemistry/holder.dm]: I _think_ this is what they are?

**Comment by Mothblocks** [comment]
Test failure isn't your fault, I'll deal with it.

File: code/modules/reagents/chemistry/holder.dm (hunk 14 above)
Comment (Thalpy) [reply code/modules/reagents/chemistry/holder.dm]: Yeah I looked it up and corrected it

**Pull request merged**